Dockerfile

```diff
@@ -18,9 +18,8 @@ COPY pkg/ pkg/
 # Build
 RUN CGO_ENABLED=0 GO111MODULE=on go build -a -o manager main.go
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+# Instead of using distroless, use Chainguard images
+FROM cgr.dev/chainguard/static:latest
```

Comment:
Is there any issue with the permission of execution so we need to replace this image?

Reply:
@Aisuko hey thanks for reaching out.

I've been following Chainguard pretty closely, and I'm been working in my spare time learning about SBOM and lower potential vulnerabilities in Docker images.

Chainguard advocates software supply chain security in this, so I'm also thinking we can at least follow what they're doing.

At the time I did this PR, Chainguard static image was not allowing other users to be used in the image.

Since then, I did find out that they have updated this to allow a certain limited range of users.

I'm still researching this, and we may or may not go this route immediately, but in the future, I would like to be able to use their images or at least a custom docker image.

Also, there is a slight discrepancies in some of the ditroless images across all meshery adapters, but it's not important now.

I still want to keep working on this.

Reply:
Thank you @acald-creator. I see. As you said their base image is professional in security. Is there any limit(e.g, open-source License)?  I saw they are a company who are in the container security industry.

Reply:
Welcome to continue work on this.

Reply:
@Aisuko Sorry for the delay in the response, but here is the link to Chainguard images at Apache 2.0 license.

https://github.com/chainguard-images/images/blob/main/LICENSE

I'm not aware of what Layer5 may have on what licenses to accept or not when using others' tools.

Reply:
@acald-creator Apache License 2.0 is quite permissive. 
@leecalcote any issue with merging this?

Reply:
@Azanul 

At the time I opened this, Chainguard had used a different type of images behind Go, and has changed since then. I will have to revisit this and see if it is useful here or on a different repo.

Reply:
> There were permissions issues because they changed the permissions for nonroot user in the static image, which could conflict with `meshery-operator` permissions.

Would you please add a reference to this change

> I will have to revisit this to see if the Dockerfile builds as expected and works properly.

👍🏻

Reply:
@Azanul it was mentioned in this conversation block, changing from `FROM gcr.io/distroless/static:nonroot` to `FROM cgr.dev/chainguard/static:latest`.

After I tracked the issue at Chainguard, they were working on updating the permissions for the static image, which has been updated since, but I have not tested this Dockerfile.

Reply:
We are ✅ with respect to licensing concerns.

Reply:
> After I tracked the issue at Chainguard, they were working on updating the permissions for the static image, which has been updated since, but I have not tested this Dockerfile.

This is the reference that I was looking for